bundles/org.openhab.binding.emotiva/src/main/java/org/openhab/binding/emotiva/internal/EmotivaProcessorHandler.java

```diff
@@ -241,37 +245,64 @@ private void startPollingKeepAlive() {
     }
 
     private void checkKeepAliveTimestamp() {
-        if (ThingStatus.ONLINE.equals(getThing().getStatusInfo().getStatus())) {
-            Optional<State> lastSeenState = state.getChannel(LAST_SEEN_STATE_NAME);
-            if (lastSeenState.isPresent()) {
-                if (lastSeenState.get() instanceof Number value) {
-                    Instant lastKeepAliveMessageTimestamp = Instant.ofEpochSecond(value.longValue());
-                    Instant deviceGoneGracePeriod = Instant.now().minus(config.keepAlive, ChronoUnit.MILLIS)
-                            .minus(DEFAULT_KEEP_ALIVE_CONSIDERED_LOST_IN_MILLISECONDS, ChronoUnit.MILLIS);
-                    if (lastKeepAliveMessageTimestamp.isBefore(deviceGoneGracePeriod)) {
+        Instant lastKeepAliveTimestamp = state.getLastSeen();
+
+        Instant deviceGoneGracePeriod = Instant.now().minus(config.keepAlive, ChronoUnit.MILLIS)
+                .minus(DEFAULT_KEEP_ALIVE_CONSIDERED_LOST_IN_MILLISECONDS, ChronoUnit.MILLIS);
+        boolean isPastGradePeriod = lastKeepAliveTimestamp.isBefore(deviceGoneGracePeriod);
+
+        if (ThingStatus.ONLINE.equals(thing.getStatusInfo().getStatus())) {
+            if (isPastGradePeriod) {
+                logger.debug(
+                        "Keep-alive job for '{}': status={}, last update to {} was '{}', past grace-period by '{}', scheduling connection retry",
+                        thing.getUID(), thing.getStatusInfo().getStatus(), LAST_SEEN_STATE_NAME, lastKeepAliveTimestamp,
```

Comment:
LAST_SEEN_STATE_NAME always has the value "no-channel#last-seen"  This is the non-existent channel, please remove.